### DISCUSSION TITLE
LibreVNA GUI on raspberry pi 5

### DISCUSSION THREAD

**Discussion by michanisani** [jankae/LibreVNA#254] [Q&A]
I install Ubuntu 24.04 LTE  (64 bits) on Raspi5
Follow the instractions of installing the GUI under Ubunto
(The instraction are in the readme file GUI Installation.)
The file [LibreVNA-GUI-Ubuntu-v1.5.1.zip] 
(https://github.com/jankae/LibreVNA/releases/download/v1.5.1/LibreVNA-GUI-Ubuntu-v1.5.1.zip)
Does not include any pre compile software.
So What I missing ?

**Comment by jankae** [comment]
I just downloaded it (by clicking on the link in your post). I am positive that it does contain the binary. There must be something going wrong with the download on your end.

**Comment by michanisani** [comment]
Correct. 
I was confuse from the Ubuntu file for LibreVNA-GUI-Ubuntu-v1.5.1.zip
This file is for X86
The file for the Pi5 is
[LibreVNA-GUI_RPi5_v1.5.1.zip](https://github.com/jankae/LibreVNA/releases/download/v1.5.1/LibreVNA-GUI_RPi5_v1.5.1.zip)
I suggest to change the file name and add X86 to the first and Ubunto to the second.
Thanks for the support.

**Comment by jankae** [comment]
Ah, yes, executing the file in LibreVNA-GUI-Ubuntu-v1.5.1.zip on a Raspberry Pi will not work.

> I suggest to change the file name and add X86 to the first

That makes sense, I may implement that for future releases.

> and Ubunto to the second.

That would only make things more confusing in my opinion. LibreVNA-GUI_RPi5_v1.5.1.zip is compiled for a Raspberry Pi but not specifically for a Raspberry Pi running Ubuntu. In fact, it is compiled on a Pi running Raspberry Pi OS.

In my mind it is still:
Ubuntu: a "real" PC running on some Intel/AMD
RPi: some Raspberry Pi running Raspbian (now named Raspberry Pi OS)

I am aware that these things are changing, especially now that there is an Ubuntu version for the RPi. But I think most linux users should be able to figure out what the problem is on their own.

**Comment by michanisani** [comment]
This mean that i can run librevna UI on rspi OS?
Great
Thanks

**Comment by jankae** [comment]
Yes, the binary for RPi will run on Raspberry Pi OS (that is also what I am using for my automated tests of the GUI/firmware)
